pkg/daemon/server/handle.go

```diff
@@ -97,9 +89,9 @@ func (cdh cniDaemonHandler) handleAdd(req *restful.Request, resp *restful.Respon
 		time.Sleep(backOffBase)
 		backOffBase = backOffBase * 2
 
-		pod, err := cdh.KubeClient.CoreV1().Pods(podRequest.PodNamespace).Get(context.TODO(), podRequest.PodName, metav1.GetOptions{})
-		if err != nil {
-			errMsg := fmt.Errorf("get pod %v/%v failed: %v", podRequest.PodName, podRequest.PodNamespace, err)
+		pod := &corev1.Pod{}
+		if err := cdh.mgrAPIReader.Get(context.TODO(), types.NamespacedName{Name: podRequest.PodName}, pod); err != nil {
```

Comment:
why use APIReader not client here>

Reply:
I think `Client` will read from the cache in local memory while `APIReader` will not. And in order to get `Pod` information from apiserver, maybe using `APIReader` is necessary here.